dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStoreIntegrationTest.java

```diff
@@ -0,0 +1,212 @@
+
+import org.hisp.dhis.IntegrationTestBase;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramService;
+import org.hisp.dhis.program.ProgramTrackedEntityAttribute;
+import org.hisp.dhis.security.acl.AccessStringHelper;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * @author Chau Thu Tran
+ */
+public class TrackedEntityAttributeStoreIntegrationTest
+    extends
+    IntegrationTestBase
+{
+    @Autowired
+    private TrackedEntityAttributeService attributeService;
+
+    private TrackedEntityAttribute attributeW;
+
+    private TrackedEntityAttribute attributeY;
+
+    private TrackedEntityAttribute attributeZ;
+
+    @Autowired
+    private TrackedEntityTypeService trackedEntityTypeService;
+
+    @Autowired
+    private ProgramService programService;
+
+    private final static int A = 65;
+
+    private final static int T = 85;
+
+    private Program programB;
+
+    @Override
+    public void setUpTest()
+    {
+
+        attributeW = createTrackedEntityAttribute( 'W' );
+        attributeW.setUnique( true );
+        attributeY = createTrackedEntityAttribute( 'Y' );
+        attributeY.setUnique( true );
+        attributeZ = createTrackedEntityAttribute( 'Z', ValueType.NUMBER );
+
+        List<TrackedEntityAttribute> attributesA = new ArrayList<>();
+        attributesA.add( attributeW );
+        attributesA.add( attributeY );
+
+        Program program = createProgram( 'A' );
+        programService.addProgram( program );
+
+        TrackedEntityType trackedEntityTypeA = createTrackedEntityType( 'A' );
+        trackedEntityTypeA.setPublicAccess( AccessStringHelper.FULL );
+        trackedEntityTypeService.addTrackedEntityType( trackedEntityTypeA );
+
+        TrackedEntityType trackedEntityTypeB = createTrackedEntityType( 'B' );
+        trackedEntityTypeB.setPublicAccess( AccessStringHelper.FULL );
+        trackedEntityTypeService.addTrackedEntityType( trackedEntityTypeB );
+
+        // Create 20 Tracked Entity Attributes (named A .. O)
+        IntStream.range( A, T ).mapToObj( i -> Character.toString( (char) i ) ).forEach( c -> attributeService
+            .addTrackedEntityAttribute( createTrackedEntityAttribute( c.charAt( 0 ), ValueType.TEXT ) ) );
+
+        // Transform the Tracked Entity Attributes into a List of
+        // TrackedEntityTypeAttribute
+        List<TrackedEntityTypeAttribute> teatList = IntStream.range( A, T )
+            .mapToObj( i -> Character.toString( (char) i ) )
+            .map( s -> new TrackedEntityTypeAttribute( trackedEntityTypeA,
+                attributeService.getTrackedEntityAttributeByName( "Attribute" + s ) ) )
+            .collect( Collectors.toList() );
+
+        // Setting searchable to true for 5 tracked entity type attributes
+        TrackedEntityTypeAttribute teta = teatList.get( 0 );
+        teta.setSearchable( true );
+        teta = teatList.get( 4 );
```

Comment:
is there a reason behind the interval/spacing between the indices `0, 4, 9, ...`? just wondering whether that should be mentioned in a comment, if there is any reason

Reply:
Just added the word "random" to the comment above. Hope that is clear enough 😆